dbt_meshify/storage/file_content_editors.py

```diff
@@ -0,0 +1,453 @@
+
+        """
+        return Path(self.node.original_file_path)
+
+    def add_model_contract(self) -> None:
+        """Adds a model contract to the model's yaml"""
+        yml_path = self.get_patch_path()
+        # read the yml file
+        # pass empty dict if no file contents returned
+        models_yml = self.file_manager.read_file(yml_path)
+
+        if isinstance(models_yml, str):
+            raise Exception(f"Unexpected string values in dumped model data in {yml_path}.")
+
+        updated_yml = self.add_model_contract_to_yml(
+            model_name=self.node.name,
+            model_catalog=self.model_catalog,
+            models_yml=models_yml,
+        )
+        # write the updated yml to the file
+        self.file_manager.write_file(yml_path, updated_yml)
+
+    def add_model_access(self, access_type: AccessType) -> None:
+        """Adds a model contract to the model's yaml"""
+        yml_path = self.get_patch_path()
+        # read the yml file
+        # pass empty dict if no file contents returned
+        models_yml = self.file_manager.read_file(yml_path)
+
+        if isinstance(models_yml, str):
+            raise Exception(f"Unexpected string values in dumped model data in {yml_path}.")
+
+        updated_yml = self.add_access_to_model_yml(
+            model_name=self.node.name,
+            access_type=access_type,
+            models_yml=models_yml,
+        )
+        # write the updated yml to the file
+        self.file_manager.write_file(yml_path, updated_yml)
+
+    def add_model_version(
+        self, prerelease: Optional[bool] = False, defined_in: Optional[os.PathLike] = None
+    ) -> None:
+        """Adds a model version to the model's yaml"""
+
+        yml_path = self.get_patch_path()
+
+        # read the yml file
+        # pass empty dict if no file contents returned
+        models_yml = self.file_manager.read_file(yml_path) or {}
+        updated_yml = self.add_model_version_to_yml(
+            model_name=self.node.name,
+            models_yml=models_yml,
+            prerelease=prerelease,
+            defined_in=defined_in,
+        )
+        # write the updated yml to the file
+        self.file_manager.write_file(yml_path, updated_yml)
+        # create the new version file
+
+        # if we're incrementing the version, write the new version file with a copy of the code
+        latest_version = int(self.node.latest_version) if self.node.latest_version else 0
+        last_version_file_name = f"{self.node.name}_v{latest_version}.{self.node.language}"
+        next_version_file_name = (
+            f"{defined_in}.{self.node.language}"
+            if defined_in
+            else f"{self.node.name}_v{latest_version + 1}.{self.node.language}"
+        )
+        model_path = self.get_resource_path()
+
+        if model_path is None:
+            raise Exception(f"Unable to find path to model {self.node.name}. Aborting.")
```

Comment:
Should there ever be a time when `get_resource_path` returns None, or do we want this to error out? My gut says erroring out is the preferred approach, that way do don't introduce surface area for defects.